internal/repo/datasource/sql_test.go

```diff
@@ -0,0 +1,321 @@
+				if !strings.Contains(err.Error(), "sql: Scan error") {
+					t.Errorf("Want: %v, Got: %v", "sql: Scan error", err)
+				}
+			},
+		},
+	}
+
+	// to execute the tests in the table
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// STEP 1: seting up all instances for the specific test case
+			db := tt.setupFunc()
+			// STEP 2: call the test function
+			rows, count, err := db.Get(tt.filter, 1, 2)
+
+			// STEP 3: validation of output
+			if tt.validator != nil {
+				tt.validator(rows, count, err)
+			}
+
+			// STEP 4: clean up/remove up all instances for the specific test case
+			if tt.cleanupFunc != nil {
+				tt.cleanupFunc()
+			}
+		})
+	}
+}
+
+//
+func TestSqlDs_Insert(t *testing.T) {
+	// table driven tests
+	tests := []struct {
+		name        string
+		tableName   string
+		data        model.Transaction
+		setupFunc   func() (sqlDs, sqlmock.Sqlmock)
+		cleanupFunc func()
+		filter      map[string]interface{}
+		validator   func(sqlmock.Sqlmock, error)
+	}{
+		{
+			name: "SUCCESS:: Insert Transaction",
+			data: model.Transaction{
+				UserId:        "1",
+				AccountNumber: 1,
+				TransactionId: "1234",
+				Amount:        1000,
+				TransferTo:    2,
+				Status:        "approved",
+				Type:          "debit",
+				Comment:       "abcd",
+			},
+			setupFunc: func() (sqlDs, sqlmock.Sqlmock) {
+				db, mock, err := sqlmock.New()
+				if err != nil {
+					t.Fail()
+				}
+				dB := sqlDs{
+					sqlSvc: db,
+					table:  "newTemp",
+				}
+				m := mock.ExpectExec(regexp.QuoteMeta("INSERT INTO newTemp(user_id, transaction_id, account_number, amount, transfer_to, status, type, comment) VALUES(?,?,?,?,?,?,?,?)")).WithArgs("1", "1234", 1, 1000.00, 2, "approved", "debit", "abcd")
+				m.WillReturnError(nil)
+				m.WillReturnResult(sqlmock.NewResult(1, 1))
+				return dB, mock
+			},
+			validator: func(mock sqlmock.Sqlmock, err error) {
+				if mock.ExpectationsWereMet() != nil {
+					t.Errorf("Want: %v, Got: %v", nil, err.Error())
+					return
+				}
+				if err != nil {
+					t.Errorf("Want: %v, Got: %v", nil, err.Error())
+					return
+				}
+			},
+		},
+		{
+			name: "FAILURE:: insert :: sql error",
+			data: model.Transaction{
+				UserId:        "1",
+				AccountNumber: 1,
+				TransactionId: "1234",
+				Amount:        1000,
+				TransferTo:    2,
+				Status:        "approved",
+				Type:          "debit",
+				Comment:       "abcd",
+			},
+			setupFunc: func() (sqlDs, sqlmock.Sqlmock) {
+				db, mock, err := sqlmock.New()
+				if err != nil {
+					t.Fail()
+				}
+				dB := sqlDs{
+					sqlSvc: db,
+					table:  "newTemp",
+				}
+				m := mock.ExpectExec(regexp.QuoteMeta("INSERT INTO newTemp(user_id, transaction_id, account_number, amount, transfer_to, status, type, comment) VALUES(?,?,?,?,?,?,?,?)")).WithArgs(sqlmock.AnyArg(), sqlmock.AnyArg(), sqlmock.AnyArg(), sqlmock.AnyArg(), sqlmock.AnyArg(), sqlmock.AnyArg(), sqlmock.AnyArg(), sqlmock.AnyArg())
```

Comment:
Looks like the data is passed to this mock. Can they be used for direct matching instead of using AnyArg()?

Reply:
done